test/e2e/kafka_binding_test.go

```diff
@@ -47,28 +47,54 @@ func testKafkaBinding(t *testing.T, messageKey string, messageHeaders map[string
 	t.Logf("Creating EventRecord")
 	eventTracker, _ := recordevents.StartEventRecordOrFail(client, loggerPodName)
 
-	t.Logf("Creating KafkaSource")
-	contribtestlib.CreateKafkaSourceOrFail(client, contribresources.KafkaSource(
-		kafkaBootstrapUrl,
-		kafkaTopicName,
-		resources.ServiceRef(loggerPodName),
-	))
-
+	t.Logf("Creating KafkaSource %s", version)
+	switch version {
+	case "v1alpha1":
+		contribtestlib.CreateKafkaSourceV1Alpha1OrFail(client, contribresources.KafkaSourceV1Alpha1(
+			kafkaBootstrapUrl,
+			kafkaTopicName,
+			resources.ServiceRef(loggerPodName),
+		))
+	case "v1beta1":
+		contribtestlib.CreateKafkaSourceV1Beta1OrFail(client, contribresources.KafkaSourceV1Beta1(
+			kafkaBootstrapUrl,
+			kafkaTopicName,
+			resources.ServiceRef(loggerPodName),
+		))
+	default:
+		t.Fatalf("Unknown KafkaSource version %s", version)
+	}
 	selector := map[string]string{
 		"topic": kafkaTopicName,
 	}
 
-	t.Logf("Creating KafkaBinding")
-	contribtestlib.CreateKafkaBindingOrFail(client, contribresources.KafkaBinding(
-		kafkaBootstrapUrl,
-		&tracker.Reference{
-			APIVersion: "batch/v1",
-			Kind:       "Job",
-			Selector: &metav1.LabelSelector{
-				MatchLabels: selector,
+	t.Logf("Creating KafkaBinding %s", version)
+	switch version {
+	case "v1alpha1":
+		contribtestlib.CreateKafkaBindingV1Alpha1OrFail(client, contribresources.KafkaBindingV1Alpha1(
+			kafkaBootstrapUrl,
+			&tracker.Reference{
```

Comment:
Is this object identical to the one declared in line 87? If so, consider declaring it before the switch and reusing it?